### PR TITLE
use a dedicated variable to force a lower compression level

### PR DESCRIPTION
**Pull request by chouquette**
### Motivation

Using the maximal compression level on armhf jobs will attempt to allocate more than 32bits binaries can address.

### Additional Notes

Related to incident-27692

### CODE REVIEW

**Comment by chouquette** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Review by alopezz — APPROVED** [review]

File: omnibus/config/projects/iot-agent.rb [hunk 1]
```diff
@@ -84,10 +84,9 @@
   COMPRESSION_THREADS = 1
 end
 
-# On armv7, dpkg is built as a 32bits application, which means
-# we can only address 32 bits of memory, which is likely to OOM
-# if we use too many compression threads or a too agressive level
-if ENV.has_key?("DEPLOY_AGENT") && ENV["DEPLOY_AGENT"] == "true" && (!ENV.has_key?("PACKAGE_ARCH") || ENV["PACKAGE_ARCH"] != "armhf")
+if ENV.has_key('FORCED_COMPRESSION_LEVEL')
```
Comment (alopezz) [inline omnibus/config/projects/iot-agent.rb:87]: I think I'd prefer a name such as this but not blocking

```suggestion
if ENV.has_key('PACKAGE_COMPRESSION_LEVEL')
```

**Review by alopezz — APPROVED** [review]

**Comment by chouquette** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by chouquette** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by chouquette** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Comment by chouquette** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable DEPLOY_AGENT=true --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

**Pull request merged**